lhotse/dataset/collation.py

```diff
@@ -163,7 +163,18 @@ def collate_audio(
     audio = torch.empty(len(cuts), first_cut.num_samples)
     if executor is None:
         for idx, cut in enumerate(cuts):
-            audio[idx] = _read_audio(cut)
+            try:
+                audio[idx] = _read_audio(cut)
+            except AudioLoadingError as e:
```

Comment:
These excepts are getting quite verbose; could you add the following code in `utils.py` and use it here and in other places?


Code: 

```python
class suppress_and_warn:
    """Context manager to suppress specified exceptions that logs the error message.

    After the exception is suppressed, execution proceeds with the next
    statement following the with statement.

         with warn_and_suppress(FileNotFoundError):
             os.remove(somefile)
         # Execution still resumes here if the file was already removed
    """

    def __init__(self, *exceptions):
        self._exceptions = exceptions

    def __enter__(self):
        pass

    def __exit__(self, exctype, excinst, exctb):
        # Returning True from __exit__ in a context manager tells Python
        # to suppress an exception.
        should_suppress = exctype is not None and issubclass(exctype, self._exceptions)
        if should_suppress:
            logging.warning(f"[Suppressed {exctype.__qualname__}] Error message: {excinst}")
        return should_suppress
```

Usage:

```python
with suppress_and_warn(AudioLoadingError, DurationMismatchError):
    audio[idx] = _read_audio(cut)
```

Reply:
Also note that the current code introduces an error, because the failed utterances are still in the batch just with zeroes. Can you convert `audio` to a list, append to it instead of assigning to specific positions, and later stack it? It's probably less efficient but I doubt it would result in a noticeable slowdown.